packages/components/src/Provider/MarigoldProvider.tsx

```diff
@@ -4,27 +4,36 @@ import {
   Theme,
   ThemeProvider,
   ThemeProviderProps,
+  cn,
   defaultTheme,
   useTheme,
 } from '@marigold/system';
 
 // Props
 // ---------------
 export interface MarigoldProviderProps<T extends Theme>
-  extends ThemeProviderProps<T> {}
+  extends ThemeProviderProps<T> {
+  className?: string;
+}
 
 // Provider
 // ---------------
 export function MarigoldProvider<T extends Theme>({
   children,
   theme,
+  className,
 }: MarigoldProviderProps<T>) {
   const outerTheme = useTheme();
   const isTopLevel = outerTheme === defaultTheme;
+  const root = theme.root?.();
 
   return (
     <ThemeProvider theme={theme}>
-      {isTopLevel ? <OverlayProvider>{children}</OverlayProvider> : children}
+      {isTopLevel ? (
+        <OverlayProvider>{children}</OverlayProvider>
+      ) : (
+        <div className={cn(root ? root : '', className)}>{children}</div>
```

Comment:
This will not get applied in regular setups, right? Because they only have only have ony provider.